core/variant/callable.cpp

```diff
@@ -327,34 +328,47 @@ Callable::operator String() const {
 	}
 }
 
+Callable Callable::create(const Variant &p_variant, const StringName &p_method) {
+	ERR_FAIL_COND_V_MSG(p_method == StringName(), Callable(), "Method argument to Callable::create method must be a non-empty string.");
+
+	switch (p_variant.get_type()) {
+		case Variant::NIL:
+			return Callable(ObjectID(), p_method);
+		case Variant::OBJECT:
+			return Callable(p_variant.operator ObjectID(), p_method);
+		default:
+			return Callable(memnew(VariantCallable(p_variant, p_method)));
+	}
+}
+
 Callable::Callable(const Object *p_object, const StringName &p_method) {
-	if (p_method == StringName()) {
+	if (unlikely(p_method == StringName())) {
 		object = 0;
-		ERR_FAIL_MSG("Method argument to Callable constructor must be a non-empty string");
+		ERR_FAIL_MSG("Method argument to Callable constructor must be a non-empty string.");
 	}
-	if (p_object == nullptr) {
+	if (unlikely(p_object == nullptr)) {
```

Comment:
I seem to remember seeing `!p_object` more often over `p_object == nullptr`, but it probably doesn't matter much.

Reply:
I see both options in the codebase and don't have a strict preference. I try to use an option depending on the context and not change it in the existing code. I think this is a case where consistency does more harm than good.